tests/src/Context/FieldContext.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\nexteuropa\Context\FieldContext.
+ */
+
+namespace Drupal\nexteuropa\Context;
+
+use Behat\Behat\Context\Context;
+
+/**
+ * Context with field management.
+ */
+class FieldContext implements Context {
+
+  /**
+   * The list of field instances before a scenario starts.
+   *
+   * @var array
+   */
+  protected $defaultFields = array();
+
+  /**
+   * The list of field groups before a scenario starts.
+   *
+   * @var array
+   */
+  protected $defaultFieldGroups = array();
+
+  /**
+   * Remember the list of fields existing by default in the site.
+   *
+   * @BeforeScenario
+   */
+  public function rememberCurrentFields() {
+    $this->defaultFields = field_info_field_map();
+  }
+
+  /**
+   * Removes any fields created by a scenario.
+   *
+   * @AfterScenario
+   */
+  public function resetFields() {
+    if (empty($this->defaultFields)) {
+      return;
+    }
+
+    $current_fields = field_info_field_map();
+    $purge_field_schema = FALSE;
+    foreach ($current_fields as $field_name => $field_info) {
+      if (!isset($this->defaultFields[$field_name])) {
+        $field_info = field_info_field($field_name);
+        if (!is_null($field_info)
+          && !$field_info['locked']
+          && !$field_info['deleted']) {
+          field_delete_field($field_name);
+          $purge_field_schema = TRUE;
+        }
+      }
+      else {
+        // We check if a test has not created an instance of an existing field.
+        // If it is the case, we delete only this instance.
+        $previous_field_info = $this->defaultFields[$field_name]['bundles'];
+        foreach ($field_info['bundles'] as $entity_type => $bundles) {
+          $default_bundles = $previous_field_info[$entity_type];
+          $created_instances = array_diff($bundles, $default_bundles);
+          if (!empty($created_instances)) {
+            foreach ($created_instances as $created_instance) {
+              $field_instance = field_info_instance($entity_type, $field_name, $created_instance);
+              field_delete_instance($field_instance);
+            }
+          }
+        }
+      }
+    }
+
+    if ($purge_field_schema) {
+      field_purge_batch(100);
+    }
+    $this->defaultFields = array();
+  }
+
+  /**
+   * Remember the list of field groups existing by default in the site.
+   *
+   * @BeforeScenario
+   */
+  public function rememberCurrentFieldGroups() {
+    $this->defaultFieldGroups = field_group_read_groups();
+  }
+
+  /**
+   * Removes any fields group created by a scenario.
+   *
+   * @AfterScenario
+   */
+  public function resetFieldGroups() {
+    if (empty($this->defaultFieldGroups)) {
+      return;
+    }
+    $current_field_groups = field_group_read_groups();
+    $this->scanFieldGroupsForResetting($current_field_groups, $this->defaultFieldGroups);
```

Comment:
I prefer my way to implement because it is more explicit, more flexible (even if it is a private method, it can be used in different cases inside the class in the future.)